### ISSUE TITLE
Writing sport output to pipe doesn't seem to work

### ISSUE THREAD

**Issue by teh** [PaulBatchelor/Sporth#3]
I'd expect the following to play without issues:

```
$ echo -e "440 0.5 sine" | sporth -o /dev/stdout | paplay 
Failed to open audio file.
```

Or have some other way of piping a wav into paplay without having to resort to raw format guessing :)

Thanks!

**Comment by PaulBatchelor** [comment]
Yup. In theory that should have worked, but it didn't. I'm guessing that this isn't a Sporth issue, but a Soundpipe one. I've pushed some fixes to the dev branch. Pull and recompile Soundpipe, then recompile Sporth. Let me know if that fixes things.

libsndfile doesn't seem to write proper wav file headers when working when the data encoding is floating point. Sox would always give warnings whenever I gave it soundpipe generated audio files, but I ignored them because it worked anyway. In Soundpipe, I have fixed sp_process in modules/base.c (the function in charge of writing audio files in Sporth) so that it writes 24-bit files instead of 32-bit float. The Sox errors went away for me, so I'm hoping it goes away for paplay as well...
